onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_helper.cc

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/shared_library/provider_api.h"
+#include "tensorrt_execution_provider.h"
+#include <iostream>
+
+namespace onnxruntime {
+
+// The newly-built graph has not yet being resolved by Graph::Resolve(), so we can't leverage ORT Graph IsInputInitializerOrOutput() API.
+// We have to do it by ourselves.
+bool TensorrtExecutionProvider::IsInputInitializerOrOutput(Graph* graph,
```

Comment:
`graph` is required so should be a reference not a pointer. Same comment applies to other functions here. If an argument cannot be nullptr, use a reference so the API clearly advertises that nullptr is not supported.

Why does the Graph need to be mutable to check if something is an input/initializer/output/outer scope/local?

Reply:
Re: Why does the Graph need to be mutable to check if something is an input/initializer/output/outer scope/local?
I will change to use immutable `ParentGraph()`.

The reason I use pointer instead of reference here is that `ParentGraph()` of graph returns pointer to `const Graph`.
And I didn't see similar function that returns a reference (I think parent graph can be null so that's probably the reason it uses pointer)

Reply:
I think it's better. As you mentioned ParentGraph can be null so that API must return a pointer, but IsInputInitializerOrOutput does not (nor should it) handle receiving a nullptr for the Graph argument, so there's no reason to make the API unclear by using a pointer for that argument. 

Put another way, the requirements for ParentGraph shouldn't dictate how a completely separate API is defined. Define it based on what the API handles/requires. 

It's trivial to support as well.

```
  const Graph* parent_graph = nullptr;
  return IsLocalValue(graph, name, subgraph_context_map) ||
         (check_ancestors && (parent_graph = graph->ParentGraph()) != nullptr &&
          IsInputInitializerOrOutput(*parent_graph, name, check_ancestors, subgraph_context_map));
```

Reply:
Modified the functions to take reference not pointer.